src/Microsoft.DotNet.Build.Tasks/PackageRestoreLogger.cs

```diff
@@ -0,0 +1,128 @@
+        }
+
+        private void ParseParameters()
+        {
+            if (!String.IsNullOrEmpty(Parameters))
+            {
+                string[] parameterPairs = Parameters.Split(';');
+                foreach (string parameterPair in parameterPairs)
+                {
+                    if (parameterPair.Length > 0)
+                    {
+                        string[] parameterNameValue = parameterPair.Split('=');
+
+                        ApplyParameter(parameterNameValue[0], parameterNameValue.Length > 1 ? parameterNameValue[1] : null);
+                    }
+                }
+            }
+        }
+
+        private void ApplyParameter(string name, string value)
+        {
+            switch (name.ToLower())
+            {
+                case "logfile":
+                    _logFile = value;
+                    break;
+                case "logtoconsole":
+                    _outputToConsole = Boolean.Parse(value);
+                    break;
+                default:
+                    // ignore unrecognized parameters
+                    break;
+            }
+        }
+
+        private void AttachToRequiredEvents(IEventSource eventSource)
+        {
+            eventSource.MessageRaised += eventSource_MessageHandler;
+        }
+
+        private void eventSource_MessageHandler(object sender, BuildMessageEventArgs e)
+        {
+            if (_packageUrlCheck(e.Message) && _packageCheck(e.Message))
+            {
+                int index = e.Message.IndexOf("https");
+                int end = e.Message.IndexOf("nupkg") + 5;
+                _packageSet.Add(e.Message.Substring(index, end - index));
+            }
+        }
+
+        private void Log(string message, StreamWriter logWriter)
+        {
+            if(logWriter != null)
+            {
+                logWriter.WriteLine(message);
+            }
+            if (_outputToConsole)
+            {
+                Console.WriteLine(message);
+            }
+        }
+
+        public void Shutdown()
+        {
+            StreamWriter logWriter = null;
+            if (!string.IsNullOrEmpty(_logFile))
+            {
+                try
+                {
+                    logWriter = new StreamWriter(new FileStream(_logFile, FileMode.Create, FileAccess.Write, FileShare.Read | FileShare.Delete, 4096, FileOptions.SequentialScan));
+                }
+                catch (IOException e)
+                {
+                    throw new InvalidOperationException("PackageRestoreLogger: Unable to create log file", e);
+                }
+            }
+            HashSet<string> feedSet = new HashSet<string>();
+            Log("Packages Restored:", logWriter);
+            foreach (var package in _packageSet)
+            {
+                Match m = Regex.Match(package, urlRegex);
+                if (m.Success)
+                {
+                    feedSet.Add(m.Groups["domain"].Value);
```

Comment:
I'd expect a regex comparison to really slow things down.  I don't know that the value of having pretty output supersedes the performance cost.